collections/_article/parallax-progress-report.md

```diff
@@ -0,0 +1,120 @@
+---
+title: "Parallax2D Progress Report"
+excerpt: "A new node is on its way to help with parallax in 2D."
+categories: ["progress-report"]
+author: "Mark DiBarry"
+image: /storage/blog/covers/progress-report-parallax2d.webp
+date: 2024-03-16 12:00:00
+---
+
+Simulating depth with backgrounds is a staple of 2D games. Godot's parallax system has been part of the core node set before Godot even went open source. With all of Godot's advancements in the last decade, the parallax system has remained (mostly) the same. In the upcoming 4.3 release, it's time for it to get some love and attention too!
+
+<video autoplay loop muted playsinline title="An example of the parallax effect">
+  <source src="/storage/blog/parallax2d/scroll.mp4" type="video/mp4">
+</video>
+
+### A fresh start
+
+`Parallax2D` is a new node intended as an all-in-one replacement for `ParallaxBackground` and `ParallaxLayer`. It's flagged as experimental in 4.3 and is subject to change while we get some more feedback, but the plan is to eventually deprecate the old nodes and make it permanent. There is feature-parity with the old system along with some new treats! You can convert your current `ParallaxBackground` and `ParallaxLayer`s by clicking on your `ParallaxBackground` node in the scene tree, then `Convert to Parallax2D` from the context dropdown.
+
+![An example of converting to Parallax2D](/storage/blog/parallax2d/button.webp)
+
+### Why the change?
+
+There've been a lot of proposals and issues regarding Godot's parallax system over the last few years. *Bogusly*, these had trouble getting movement without a significant overhaul of the two nodes that it's comprised of: `ParallaxBackground` and `ParallaxLayer`. Godot has introduced so many new features and upgrades in related areas (especially with 4.0) that the parallax system didn't originally need to support (because they didn't exist; A reasonable justification indeed!). I recently ran into one such case working on a faux-lighting system to replicate the look of lights on early video game consoles.
```

Comment:
I guess if I had to break it down, "Bogus" was a common phrase in the 90's meaning "disappointing", but is just used here to comedic effect (since it sounds very dated). It's just my writing style, but I can revise the post to remove things that don't convey a more serious tone.

If I had to pin-point the era for lighting in "early video game consoles", I'd say it'd be the 3rd (NES) through the 4th (SNES, Genesis, Gameboy). Before the 3rd, it was difficult to display much detail, and from somewhere between the 5th and 6th on they were able to use real lighting effects.

Reply:
> I can revise the post to remove things that don't convey a more serious tone.

I'm not a super fan of the style, but I think it's OK to leave most of them in! I just dislike the ones that don't make sense. I'd be happy with "Disappointedly".

I think it would be better to say "NES-era" consoles, rather than "early". Early to me is Atari 2600!

Reply:
It's okay - My style's not for everyone! Not a big deal.

I have seen it in the 3rd gen, but it was definitely more common in the 4th. I'll split the difference. Thanks for the feedback!

Reply:
Actually, now that you mention it, should I just remove references to my personal project? It's not 100% necessary to get the concept across, and digresses from the blog post early on.

Reply:
To be honest, I did get a bit lost during the explanation of your project and the problems you found. It might have been clearer with screenshots, but that would make the article even longer. If you can manage without the digression (while still trying to make it clear why the old system had limitations), I think it would indeed improve the article.